src/commands/profiles/exist.ts

```diff
@@ -42,26 +44,29 @@ export default class ProfilesExist extends Command {
       );
       cli.action.stop();
 
-      cli.action.start("Querying profiles...");
       const client = new cosmos.CosmosClient({ endpoint, auth: { key } });
       const database = client.database("agid-documentdb-test");
       const container = database.container("profiles");
 
-      // retrieve all fiscalCodes from database
-      const response = container.items.query(
-        "SELECT c.fiscalCode FROM c WHERE c.version = 0",
-        {
-          enableCrossPartitionQuery: true
-        }
-      );
-      const result = (await response.toArray()).result;
-      cli.action.stop();
-      if (result === undefined) {
-        this.error("No result");
-        return;
-      }
-      // create a set of fiscalCodes
-      const allProfilesCf = new Set<string>(result.map(r => r.fiscalCode));
+      const hasProfile = async (fiscalCode: string): Promise<boolean> => {
+        const documentId = `${fiscalCode}-${"0".repeat(16)}`;
+        const response = container.items.query(
+          {
+            parameters: [
+              {
+                name: "@documentId",
+                value: documentId
+              }
+            ],
+            query: "SELECT VALUE COUNT(1) FROM c WHERE c.id = @documentId"
+          },
+          {
+            enableCrossPartitionQuery: true
```

Comment:
if you provide the partitionKey as well (it's fiscalCode) you will not need this anymore - basically, with `enableCrossPartitionQuery: true`, cosmosDb will issue the query to all partitions (much more expensive)

Reply:
I tried removing that option but it gives me a 400 error (BadRequest)
![Schermata 2019-07-12 alle 16 54 14](https://user-images.githubusercontent.com/822471/61137397-b5f8ba00-a4c5-11e9-9304-c6d545bc3c87.png)
I changed the query putting fiscalCode as WHERE condition and it works (maybe because it is the partitionKey)
In this way it seems to go even faster

Reply:
yes hitting the right partition directly is much faster than querying all partitions!!!

Reply:
Ok, thanks for the explanation. The new changes are already committed